pkg/server/server_sql.go

```diff
@@ -996,6 +1015,9 @@ func (s *SQLServer) preStart(
 		return err
 	}
 
+	// Create and start the span config reconciliation job if none exist.
+	_ = s.execCfg.StartSpanConfigReconciliationJobHook(ctx)
```

Comment:
Also, doesn't this need version gating? We can't run this until after we've activated a version where all of the nodes know about this job type.